### PR TITLE
Update test docs for OTel Demo

### PR DESCRIPTION
**Pull request by danielbdias**
Hi folks!

On this PR, I'm updating the OTel Demo testing docs to show the recent changes that we did there.

---

**Preview**: https://deploy-preview-3173--opentelemetry.netlify.app/docs/demo/tests/

### CODE REVIEW

File: content/en/docs/demo/tests.md [hunk 1]
```diff
@@ -1,16 +1,26 @@
...
 
 Currently, the repository includes E2E tests for both the frontend and backend
 services. For the Frontend we are using [Cypress](https://www.cypress.io/)
 execute the different flows in the web store. While the backend services use
-[AVA](https://avajs.dev) as the main testing framework.
+[AVA](https://avajs.dev) as the main testing framework for integration tests and
+[Tracetest](https://tracetest.io/) for trace-based tests.
 
-To run the test you can simply run `make run-tests` at the root directory.
+To run all the tests you can simply run `make run-tests` at the root directory.
```
Comment (chalin) [inline content/en/docs/demo/tests.md:12]:
```suggestion
To run all the tests, execute `make run-tests` from the root directory.
```

File: content/en/docs/demo/tests.md [hunk 2]
```diff
@@ -1,16 +1,26 @@
...
-In case you need to run a specific suite of tests you can execute
-`docker compose run frontendTests`[^1] for the frontend tests or
-`docker compose run integrationTests`[^1] for the backend tests.
+Otherwise, if you want to run a specific suite of tests you can execute specific
+commands for each type of test:
+
+- **Frontend tests**: `docker compose run frontendTests`[^1]
+- **Backend tests (integration)**: `docker compose run integrationTests`[^1]
+- **Backend tests (trace-based tests)**:
+  `docker compose run traceBasedTests`[^1]
```
Comment (chalin) [inline content/en/docs/demo/tests.md:20]: How about putting the footnote at the end of the opening sentence instead? 

```suggestion
commands for each type of test[^1]:

- **Frontend tests**: `docker compose run frontendTests`
- **Backend tests**:
  - Integration: `docker compose run integrationTests`
  - Trace-based: `docker compose run traceBasedTests`
```

File: content/en/docs/demo/tests.md [hunk 3]
```diff
@@ -1,16 +1,26 @@
...
+
+To know more details about these tests you can see them on the official
+repository,
+[here](https://github.com/open-telemetry/opentelemetry-demo/tree/main/test).
```
Comment (chalin) [inline content/en/docs/demo/tests.md:24]:
```suggestion
To learn more these tests, see 
[Service Testing](https://github.com/open-telemetry/opentelemetry-demo/tree/main/test).
```

**Review by chalin — COMMENTED** [review]
See inline comments, thanks.

**Comment by danielbdias** [comment]
@chalin it makes sense, thanks!

**Review by cartermp — APPROVED** [review]

**Review by mviitane — APPROVED** [review]

**Review by chalin — APPROVED** [review]

**Pull request merged**